dear_petition/petition/etl/email.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from dear_petition.users.models import User
+from dear_petition.petition.etl.load import import_ciprs_records
+
+from dear_petition.sendgrid.models import Email
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_username_and_label(addr):
+    """
+    Extracts email into parts for importing.
+
+    For example:
+        first.last+mylabel@example.com -> first.last, mylabel
+    """
+    email_username = addr.split("@")[0]
+    username_parts = email_username.split("+")
+    username = username_parts[0]
+    try:
+        label = username_parts[1]
+    except IndexError:
+        label = ""
+    return username, label
+
+
+def sendgrid_webhook_listener(email_id):
+    """
+    Associate email attachments to a user and pass off to the ETL process.
+    """
+    email = Email.objects.get(id=email_id)
+    username, label = extract_username_and_label(email.recipient)
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        logger.error(f"Username {username} not found for email '{email}'")
+        return
+    attachment_files = []
+    for attachment in email.attachments.all():
+        attachment_files.append(attachment.file)
+    batch = import_ciprs_records(
+        files=attachment_files, user=user, parser_mode=1, batch_label=label
+    )
```

Comment:
The cool thing about using the email listener is that we can addsome some background processing on the ciprs record here without worrying about taking too much time!

If you agree it's a good idea, I think it'd be nice to open an issue detailing potential improvements. One I've thought about it adding image PDF parsing. Another is using something called Camelot which is much slower than xpdfreader but it has better support for parsing of tables.